src/document.rs

```diff
@@ -208,51 +273,135 @@ fn extract_value_single_or_list(any: &PyAny) -> PyResult<Vec<Value>> {
     }
 }
 
-#[pymethods]
-impl Document {
-    #[new]
-    #[pyo3(signature = (**kwargs))]
-    fn new(kwargs: Option<&PyDict>) -> PyResult<Self> {
-        let mut document = Document::default();
-        if let Some(field_dict) = kwargs {
-            document.extend(field_dict)?;
+fn extract_value_single_or_list_for_type(
+    any: &PyAny,
+    field_type: &tv::schema::FieldType,
+    field_name: &str,
+) -> PyResult<Vec<Value>> {
+    // Check if a numeric fast field supports multivalues.
+    let allow_multivalues = match field_type {
+        tv::schema::FieldType::U64(opts)
+        | tv::schema::FieldType::I64(opts)
+        | tv::schema::FieldType::F64(opts) => opts
+            .get_fastfield_cardinality()
```

Comment:
I think this will not work after upgrading this crate to Tantivy 0.20? Probably best to consider to #82.

Reply:
Ack. If that change is landing soon, I can wait to rebase changes on top of that.

Reply:
I've now rebased this on top of #82 .

Reply:
@GodTamIt I have noticed the review request and have this on my to-do list, but I wanted to wait until the bump to Tantivy 0.20 had landed, which is the case now. So I'll do this as soon as I find the time.